app/src/test/js/keysCtrlTest.js

```diff
@@ -0,0 +1,158 @@
+describe(
+        'SyncKeysCtrl and SearchKeysCtrl',
+        function() {
+
+            beforeEach(module('esbMessageAdminApp'));
+
+            var syncKeysCtrl, searchKeysCtrl, http, globals, scope, rootScope, esbMessageService, filter;
+
+            beforeEach(inject(function($rootScope, $controller, $httpBackend,
+                    Globals, EsbMessageService, $filter) {
+                rootScope = $rootScope;
+                syncKeysScope = $rootScope.$new();
+                searchKeysScope = $rootScope.$new();
+                http = $httpBackend;
+                globals = Globals;
+                esbMessageService = EsbMessageService;
+                filter = $filter;
+
+                (syncKeysCtrl) = $controller('SyncKeysCtrl', {
+                    '$scope' : syncKeysScope
+                });
+                http.expectGET("api/key/tree/Entities").respond(
+                        syncKeysSuccessResponse);
+                http.flush();
+
+                (searchKeysCtrl) = $controller('SearchKeysCtrl', {
+                    '$scope' : searchKeysScope
+                });
+                http.expectGET("api/key/tree/SearchKeys").respond(
+                        searchKeysSuccessResponse);
+                http.flush();
+
+            }));
+
+            function loadTest(scope, response) {
+                expect(scope.entities || scope.searchKeys).toEqual(
+                        response.tree);
+                expect(scope.parent).toEqual(response.tree);
+            }
+
+            it('loads Sync and Search Keys Tab', function() {
```

Comment:
Not a huge deal, but personally it makes more sense to me to have the `it('should do the thing')` first, then the methods that support it. When the `loadTest` function is first, it's like, "Okay, where is this used?" then you have to scroll down and back up, same for `crumbTest`. If it were the other way around, it's clear what is happening.